### ISSUE TITLE
Self reference FK breaks relations (TableFieldsGenerator.php:482)

### ISSUE THREAD

**Issue by randycasburn** [InfyOmLabs/laravel-generator#366]
Given | is
--|--
table| employees |
fields| id, reportsTo
FK_employee_reportsTo | **employees**.id -> **employees**.reportsTo

id  | reportsTo
---|----------
2  |	2
1  |	2
3  |	2
4  |	2
5  |	2
8  |	2
6  |	5
7  |	5
9  |	5

**Problem**: Laravel THROWS an _index not found exception_
**in**: TableFieldsGenerator.php  detectManyToOne()

This statement:
 
`            if ($foreignField == $tables[$foreignTable]->primaryKey) {
`

around line 492 expects the $foreignTable to be in the $tables array. The $table array has not been decorated with this table yet. A new Table reference is being constructed and this is a self referential FK to that same Table. Therefore this _if statement_ THROWS an _index not found exception_. This may have implications elsewhere.

My easy fix was to replace this statement:
 
`            if ($foreignField == $tables[$foreignTable]->primaryKey) {
`

with this:

```
if(!array_key_exists($foreignTable, $tables)) {
  $primaryKey = $modelTable->primaryKey;
} else {
  $primaryKey = $tables[$foreignTable]->primaryKey;
}
if ($foreignField == $primaryKey) {

```

**Comment by randycasburn** [comment]
Additionally, the Model's method name will need to be addressed/changed too.  With my code change, the self reference relationship method in the model is named 'employee' rather than the expected 'reportsTo'.

**Comment by mitulgolakiya** [comment]
@randycasburn fixed in the latest version after merging #616. For better results, I would recommend using `reports_to_id` field then it will give better results.
